chia/rpc/wallet_rpc_api.py

```diff
@@ -1706,6 +1706,24 @@ async def nft_get_info(self, request: Dict) -> EndpointResult:
                     "success": False,
                     "error": f"Launcher coin record 0x{uncurried_nft.singleton_launcher_id.hex()} not found",
                 }
+            # Get minter DID
```

Comment:
is `nft_get_info` call only called when viewing offers? method is pretty generic, so think it's better to add an additional parameter to request did minter, like `get_minter_did: bool` as it can be pretty expensive.

Reply:
Right, the GUI calls `nft_get_info` almost exclusively in the context of creating/viewing offers.

We want to show the minter in the offer summary as well as on the NFT Detail screen. For the NFT Detail screen, we use the data from `nft_get_nfts`.

Would it be possible to store the minter DID when a coin is added so that `nft_get_nfts` doesn't have to fetch it on-demand?

Reply:
np

Reply:
@ytx1991 in that case we need to fetch minter DID when NFTs are added and store it in the DB 

what's important is to not block syncing as it's slow enough, so should happen after the sync or in parallel (using a thread)

Reply:
Don't think it will be a problem, since we fetch coin data several times when adding a NFT. It's just one more fetch, the time complexity is still O(1).

Reply:
it's not about wallet's time complexity, it's about fullnode's time complexity.